nntrainer/include/tensor.h

```diff
@@ -466,15 +468,30 @@ class Tensor {
    */
   int setDim(TensorDim d);
 
+  /**
+   * @brief     return if current tensor is contiguous, if not, you can't write
+   *            on this tensor
+   * @retval    bool is contigous
+   */
+  const bool isContiguous() const noexcept { return _is_contiguous; }
```

Comment:
can you elaborate on the use of this variable? 
Also, when is this false as all the constructors set them `true`.

Reply:
It will be `false` when non-contiguous tensor is available. (such as broadcasted Tensor, or read-only transposed tensor)

I am planning to make `data` a shared pointer and reference it in diffre

While, I could've added more commit to show it's usage, I chose to split PR to prevent `n commits/1 PR`

Reply:
Got it.